ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/git/GitServiceClientImpl.java

```diff
@@ -806,129 +760,137 @@ public void pull(DevMachine devMachine,
         return asyncRequestFactory.createPostRequest(url, pullRequest).send(dtoUnmarshallerFactory.newUnmarshaller(PullResponse.class));
     }
 
-    /** {@inheritDoc} */
     @Override
     public void diff(DevMachine devMachine,
                      ProjectConfigDto project,
                      List<String> fileFilter,
-                     DiffRequest.DiffType type,
+                     DiffType type,
                      boolean noRenames,
                      int renameLimit,
                      String commitA,
                      String commitB, @NotNull AsyncRequestCallback<String> callback) {
-        DiffRequest diffRequest = dtoFactory.createDto(DiffRequest.class)
-                                            .withFileFilter(fileFilter)
-                                            .withType(type)
-                                            .withNoRenames(noRenames)
-                                            .withCommitA(commitA)
-                                            .withCommitB(commitB)
-                                            .withRenameLimit(renameLimit);
+        StringBuilder params = new StringBuilder().append("?projectPath=").append(project.getPath());
```

Comment:
Can we reuse code for method with callback and with promise? mb we can create some private method `AsyncRequest diff(...)` which will construct needed request. And then method with promise will have following code
```
    return diff(...).send(new StringUnmarshaller())
```
And method with callback:
```
  diff(...).send(callback);
```